lms/product/d2l/_plugin/course_copy.py

```diff
@@ -33,7 +33,7 @@ def find_matching_group_set_in_course(self, course, group_set_id):
     @classmethod
     def factory(cls, _context, request):
         return cls(
-            request.find_service(D2LAPIClient),
```

Comment:
Switching to name based registration on these. Main reason to avoid having to deal with new and existing circular dependencies.

Reply:
What are the specific circular dependencies that you are trying to work around? So far my understanding is that the main reason for having name vs interface lookup was to do with when the code was written (older code uses name, newer code uses interfaces). Are there other reasons for favoring one over the other?

Reply:
Also the name here implies that the service is called `D2LService` rather than `D2LAPIClient`.